wallet/wallet-cli/src/config.rs

```diff
@@ -0,0 +1,152 @@
+// Copyright (c) 2023 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::{
+    net::SocketAddr,
+    path::{Path, PathBuf},
+    str::FromStr,
+};
+
+use clap::{Parser, Subcommand};
+use common::chain::config::ChainType;
+use utils::default_data_dir::{default_data_dir_for_chain, prepare_data_dir};
+
+use crate::WalletCliError;
+
+#[derive(clap::ValueEnum, Clone, Debug)]
+pub enum Network {
+    Mainnet,
+    Testnet,
+    Regtest,
+    Signet,
+}
+
+#[derive(Parser, Debug)]
+pub struct WalletCliArgs {
+    /// Network
+    #[arg(long, value_enum, default_value_t = Network::Mainnet)]
+    pub network: Network,
+
+    /// Optional path to the directory containing the wallet files
+    #[clap(long)]
+    pub wallets_dir: Option<PathBuf>,
+
+    /// Optional wallet file name
+    #[clap(long)]
+    pub wallet_name: Option<PathBuf>,
+
+    /// RPC address
+    #[clap(long)]
+    pub rpc_address: Option<SocketAddr>,
+
+    /// Path to the RPC cookie file. If not set, the value is read from the default cookie file location.
+    #[clap(long)]
+    pub rpc_cookie_file: Option<String>,
+
+    /// RPC username
+    #[clap(long)]
+    pub rpc_username: Option<String>,
+
+    /// RPC password
```

Comment:
RPC password (either provide a username and password, or use a cookie file. You cannot use both)